src/test/java/org/commcare/formplayer/tests/MultiSelectCaseClaimTest.java

```diff
@@ -0,0 +1,113 @@
+import java.util.List;
+
+/**
+ *
+ */
+@WebMvcTest
+@ContextConfiguration(classes = TestContext.class)
+public class MultiSelectCaseClaimTest extends BaseTestClass {
+
+    private static final String APP_NAME = "case_claim_with_multi_select";
+    @Autowired
+    CacheManager cacheManager;
+
+    @Captor
+    ArgumentCaptor<String> urlCaptor;
+
+    @Captor
+    ArgumentCaptor<MultiValueMap<String, String>> requestDataCaptor;
+
+    @Override
+    @BeforeEach
+    public void setUp() throws Exception {
+        super.setUp();
+        configureRestoreFactory("caseclaimdomain", "caseclaimusername");
+        cacheManager.getCache("case_search").clear();
+    }
+
+    @Override
+    protected String getMockRestoreFileName() {
+        return "restores/caseclaim.xml";
+    }
+
+    @Test
+    public void testCaseClaimWithMultiSelectList() throws Exception {
+        configureQueryMock();
+        configureSyncMock();
+
+        // default search is on so we should skip to search results directly
+        EntityListResponse entityResp = sessionNavigateWithQuery(new String[]{"1", "action 1"},
+                APP_NAME,
+                null,
+                EntityListResponse.class);
+        assertTrue(entityResp.isMultiSelect());
+
+        String[] selectedValues =
+                new String[]{"94f8d030-c6f9-49e0-bc3f-5e0cdbf10c18", "0156fa3e-093e-4136-b95c-01b13dae66c7",
+                        "0156fa3e-093e-4136-b95c-01b13dae66c8"};
+        String[] selections = new String[]{"1", "action 1", MultiSelectEntityScreen.USE_SELECTED_VALUES};
+        CommandListResponseBean commandResponse = sessionNavigateWithQuery(selections,
+                APP_NAME,
+                null,
+                selectedValues,
+                CommandListResponseBean.class);
+
+        // `use_selected_values' should be replaced in returned selections
+        Assertions.assertNotEquals(selections, commandResponse.getSelections());
+
+        // Verify case claim request
+        verify(webClientMock, times(1)).post(urlCaptor.capture(), requestDataCaptor.capture());
+        assertEquals("http://localhost:8000/a/test/phone/claim-case/", urlCaptor.getAllValues().get(0));
+        MultiValueMap<String, String> requestData = requestDataCaptor.getAllValues().get(0);
+
+        // cases that are owned should not be in the claim request
+        List<String> casesToBeClaimed = Arrays.asList("0156fa3e-093e-4136-b95c-01b13dae66c7",
+                "0156fa3e-093e-4136-b95c-01b13dae66c8");
+        assertEquals(requestData.get("case_id"), casesToBeClaimed);
+    }
+
+    private void configureSyncMock() {
+        when(webClientMock.post(anyString(), any()))
+                .thenReturn("");
+    }
+
+    private void configureQueryMock() {
+        when(webClientMock.postFormData(anyString(), any(Multimap.class)))
+                .thenReturn(FileUtils.getFile(this.getClass(),
+                        "query_responses/case_claim_multi_select_response.xml"));
```

Comment:
It's generally better practice to configure the mock to do the validation instead of doing it with captors. There's an example of that here: https://github.com/dimagi/formplayer/blob/a98a768874b9c32911566e61a048ad5d3f519318/src/test/java/org/commcare/formplayer/tests/EndOfFormNavFormLinkingWithQueryTests.java#L59-L72

Also the case claim endpoint doesn't respond with any content, just a status code of 201.

Reply:
I looked into this though it turned out it would not be a strict enough test for the claim request. The reason being even if we don't mock the case claim request the test will pass without any failure as we don't really use the return values from the [claim request](https://github.com/dimagi/formplayer/blob/master/src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java#L486) unlike the case query request. So configuring the mock behaviour won't guarantee that the webClient was fired with those specific arguments only and not some other arguments unless we also configure the mock to throw an exception when the arguments do not match. 

I also like the argument capture style in this case since it's more assertive. This test's sole purpose is to check for the correct claim request and making assertions on argument conveys that much better.